content/doc/developer/architecture/web.adoc

```diff
@@ -0,0 +1,23 @@
+---
+title: Web Framework
+layout: developersection
+summary: How Jenkins uses the Stapler framework
+---
+
+== Request Handling
+
+Jenkins classes are bound to URLs by using http://stapler.kohsuke.org[Stapler]. The singleton +jenkinsdoc:Jenkins[]+ instance is bound to the context root (i.e. "/") URL, and the rest of the objects are bound according to their reachability from this root object. Stapler uses reflection to recursively determine how to process any given URL. A few examples of how the URL `/foo/bar` could be processed:
```

Comment:
Unfortunately not, only core. https://github.com/jenkins-infra/jenkins.io/blob/611c56e050accfd48319ef3b35d551b78838d561/STYLEGUIDE.adoc#javadoc-links

Can probably be added for plugins as well (but probably only in the simple `javadoc`/`staplerdoc` forms, and specifying the plugin ID. We have http://javadoc.jenkins.io/plugin/ now, after all.